packages/astro/src/vite-plugin-astro-server/plugin.ts

```diff
@@ -51,13 +54,33 @@ export default function createVitePluginAstroServer({
 			const localStorage = new AsyncLocalStorage();
 
 			/** rebuild the route cache + manifest */
-			async function rebuildManifest() {
+			async function rebuildManifest(path: string | null = null) {
 				pipeline.clearRouteCache();
-				routeManifest = injectDefaultDevRoutes(
-					settings,
-					devSSRManifest,
-					await createRouteManifest({ settings, fsMod }, logger), // TODO: Handle partial updates to the manifest
-				);
+
+				// If a route changes, we check if it's part of the manifest and check for its prerender value
+				if (path !== null) {
+					const route = routeManifest.routes.find((r) => path.endsWith(r.component));
+					if (!route) {
+						return;
+					}
+					if (route.type !== 'page' && route.type !== 'endpoint') return;
+
+					const routePath = fileURLToPath(new URL(route.component, settings.config.root));
+					if (!fsMod.existsSync(routePath)) {
+						// Route was renamed so it does not exist anymore
+						return;
+					}
```

Comment:
Shouldn't the manifest be updated in this case so it knows about the new name and stops handling the old one?

Reply:
I don't think so. Basically when there's a rename, I think `unlink` and `add` are triggered as well. The comment is not super clear, I'll clarify

Reply:
I checked and this actually doesn't happen